src/js/components/DateInput/DateInput.js

```diff
@@ -62,6 +65,21 @@ const DateInput = forwardRef(
       initialValue: defaultValue,
     });
 
+    let timestamp = false;
+    if (Array.isArray(defaultValue)) {
```

Comment:
could pull the array vs string out into a function and simplify this to a simple defaultValue vs value decision.